src/pobox.erl

```diff
@@ -40,16 +68,36 @@
 -export_type([max/0, filter/0, in/0, mail/0, note/0]).
 
 -record(state, {buf :: buffer(),
-                owner :: pid(),
+                owner :: name(),
+                owner_pid :: pid(),
+                owner_monitor_ref :: undefined | reference(),
                 filter :: undefined | filter(),
-                filter_state :: undefined | term()}).
-
--export([start_link/3, start_link/4, start_link/5, 
-        resize/2, active/3, notify/1, post/2, post_sync/3]).
+                filter_state :: undefined | term(),
+                heir :: undefined | pid() | atom(),
+                heir_data :: undefined | term(),
+                name :: name()}).
+
+-record(pobox_opts, {
+  name :: undefined | name(),
+  owner = self() :: name(),
+  size = 100 :: non_neg_integer(),
```

Comment:
Done, I return badarg now. I didn't change the `start_link` functions params because it's possible this map could be read from a configuration which would require addional code to extract. I am leaning towards convenience.